plugins/config.go

```diff
@@ -41,47 +43,148 @@ func (pc *registarConfig) RegisterLOOP(loopID string, cmdName string) (func() *e
 	return cmdFn, pc.grpcOpts, nil
 }
 
-// EnvConfig is the configuration interface between the application and the LOOP executable. The values
-// are fully resolved and static and passed via the environment.
+// EnvConfig is the configuration interface between the application and the LOOP executable.
+// The values are fully resolved and static and passed via the environment.
 type EnvConfig interface {
 	PrometheusPort() int
+
+	// Enables trace generation and collection on the plugin
+	TracingEnabled() bool
+
+	// The logical address of the trace collector
+	TracingCollectorTarget() string
+
+	// Attributes to be added to the node's tracing context
+	TracingAttributes() map[string]string
+
+	// The sampling ratio for the node's tracing context
+	TracingSamplingRatio() float64
 }
 
 // SetCmdEnvFromConfig sets LOOP-specific vars in the env of the given cmd.
+// It also, due to the plugin library, forwards all env vars from the host automatically.
+// TODO: BCF-2662: Remove once we can use skipHostEnv in hashicorp/go-plugin v1.5.0
+// This method is consumed by the host.
 func SetCmdEnvFromConfig(cmd *exec.Cmd, cfg EnvConfig) {
-	forward := func(name string) {
-		if v, ok := os.LookupEnv(name); ok {
-			cmd.Env = append(cmd.Env, name+"="+v)
+	injectEnv := map[string]string{
+		"CL_PROMETHEUS_PORT":       strconv.Itoa(cfg.PrometheusPort()),
+		"TRACING_ENABLED":          strconv.FormatBool(cfg.TracingEnabled()),
+		"TRACING_COLLECTOR_TARGET": cfg.TracingCollectorTarget(),
+		"TRACING_SAMPLING_RATIO":   strconv.FormatFloat(cfg.TracingSamplingRatio(), 'f', -1, 64),
+	}
+
+	for k, v := range cfg.TracingAttributes() {
+		injectEnv["TRACING_ATTRIBUTE_"+k] = v
+	}
+
+	for k, v := range injectEnv {
+		cmd.Env = append(cmd.Env, k+"="+v)
+	}
+}
+
+// isTracingEnabled parses and validates the TRACING_ENABLED environment variable.
+func getTracingEnabled() (bool, error) {
+	tracingEnabledString := os.Getenv("TRACING_ENABLED")
+	if tracingEnabledString == "" {
+		return false, nil
+	}
+	return strconv.ParseBool(tracingEnabledString)
+}
+
+// getValidCollectorTarget validates TRACING_COLLECTOR_TARGET as a URL.
+func getValidCollectorTarget() (string, error) {
+	tracingCollectorTarget := os.Getenv("TRACING_COLLECTOR_TARGET")
+	_, err := url.ParseRequestURI(tracingCollectorTarget)
+	if err != nil {
+		return "", fmt.Errorf("invalid TRACING_COLLECTOR_TARGET: %w", err)
+	}
+	return tracingCollectorTarget, nil
+}
+
+// getTracingAttributes collects attributes prefixed with TRACING_ATTRIBUTE_.
+func getTracingAttributes() map[string]string {
+	tracingAttributes := make(map[string]string)
+	for _, env := range os.Environ() {
+		if strings.HasPrefix(env, "TRACING_ATTRIBUTE_") {
+			tracingAttributes[strings.TrimPrefix(env, "TRACING_ATTRIBUTE_")] = os.Getenv(env)
 		}
 	}
-	forward("CL_LOG_SQL_MIGRATIONS")
```

Comment:
Was this removed intentionally? We'll have to add it back soon.

Reply:
Yea, I removed it intentionally since all host env vars are forwarded AFAIK.

Reply:
Yeah, but we want to use `SkipHostEnv` to avoid that: https://smartcontract-it.atlassian.net/browse/BCF-2662